### ISSUE TITLE
painful slow loading works fine with VPN  ... ISP?

### ISSUE THREAD

**Issue by tidder99** [yuliskov/SmartTube#1893]
randomly slow loading of menu minutes at a time problems with login in
almost like the connection is being randomly dropped 
but no issue on the other apps or default youtube 
i'm in the uk
- used different DNS
- defaulted back to isp router from pfsense
- uninstalled and reinstalled app stable and beta old and new
- ran it on multiple fire tv 4K sticks

what worked was running it through a VPN 
it must be my ISP Virgin Media 
but the funny thing is everything else works including the YouTube but not this app
I'm guessing something might be blocked at the ISP level

I guess there is nothing which  the developer can do unless it's a funny bug to do with TTL something like that
just hope the Post help somebody else
if anybody needs more info feel free to post

**Comment by toastyjc** [comment]
I am having the exact same problem, same isp (virgin media) in the UK.

Absolutely no idea what's going on.

**Comment by tidder99** [comment]
yeah really annoying since I installed the app a couple weeks ago been steadily getting worse
luckily I already had a vpn
there's free vpns like "Proton VPN Free" I can't say how good they are though

**Comment by toastyjc** [comment]
I have tried to build the app myself in Android Studio but whenever I debug the app it just exits immediately, I have a very limited knowledge of java and have spent a few too many hours trying to work out what is causing it to exit immediately after it has deployed into a virtual environment.

I am just trying to run a network profile to see exactly what requests are delayed, can confirm everything working fine with a VPN, however, it isn't a solution as I don't typically have a VPN.

Can anyone post an idea of what urls exactly the app is communicating with when navigating?
This will help me when I inevitably call my ISP to work out what is even going on

**Comment by tidder99** [comment]
ran the app through a proxy with and  without vpn
here is the logs if it's helpful
[Not_working_tinyproxy.log](https://drive.google.com/file/d/1oWcuNEh2CEZk9V9fl8e5xHugFoq32nIU/view?usp=sharing)
[VPN_working_tinyproxy.log](https://drive.google.com/file/d/1Iar2S2D_h9Wz2n0pxWag6iNDn9lpe27a/view?usp=sharing)

**Comment by awfullyterrible** [comment]
Can confirm the same issue, ISP is Virgin in the UK. 4k Firestick.
VPN has nearly sorted it but it’s still slower than before. Odd.

**Comment by saitrix** [comment]
Can confirm as well, same issue with Virgin Media.

**Comment by Roosterno1** [comment]
Yup, virgin too.....   Is unusable via winscribe as well.
my hub is in modem mode, plus I'm using DOH on my router, so something is afoot  😢

**Comment by TonyHoyle** [comment]
Looking at that dump it's just hitting youtube.com as you'd expect, with some early calls to github (checking for new versions?).   Likely they're doing some kind of DPI that's blocking the app explicitly.

I expect it's not particularly sophisticated and changing the app to look 'more like' the official one would be a doable thing.  Alas don't have the time with my day job to experiment..

I just routed my ipv6 connection through the shield and it started working.. so they're only doing the DPI over v4.  Not an option for everyone though.

**Comment by toastyjc** [comment]
@TonyHoyle I dug through the code whilst trying to get it to run and debug the networking (was unsuccessful) however from what I could see it simply connects to YouTube via a normal http get request using YouTube.com/TV spoofed as a referrer, and a Samsung TV as the device.

I'm not an expert on networking, however, I'd be amazed if Virgin Media has managed to implement something to block/throttle what is essentially a http request masquerading as the YouTube app on a Samsung TV.

Really interesting you have it working via ipv6. Wasn't aware VM were handing those out (at least on my residential line, anyway!) Would be interested in hearing more about how you've managed that if you have the time to share?

**Comment by TonyHoyle** [comment]
ipv6 is via AAISP which carries my work traffic, although there is ipv6 traffic on VMs network in certain areas (not mine, as far as I can tell) - that's been covered in various places.

Blocking is common in the UK because the government mandated that the larger ISPs do it.  DPI can be quite sophisticated too.  Not sure the more complex stuff would scale to ISP level but it's quite possible to fingerprint an app like Smart Youtube.

To really nail it you'd have to compare packet traces with standard youtube.. might be quite subtle differences.   It's also worth recompiling with a different referer... although it's https so unless they're doing an MITM (unlikely, lots of people would notice the certificate change) I'm not sure that is going to help.

I'm also waiting for FTTP - VM is mostly an interim thing for me, then I'll switch both lines to AAISP.

**Comment by tidder99** [comment]
for me it feels like a fault 
definitely is disturbing if they can block apps and not tell anybody 
especially when it's such a minor app just designed to block ads
what's a stop someone like Amazon sliding them some money to make their competitors go away

**Comment by TonyHoyle** [comment]
blocked.org.uk used to be a good site to see the sheer randomness of ISP blocking but it's a bit broken these days... the saved search lists still work though and you can get the idea.  They really don't have to tell anyone about a block - in fact for some classes of block (court mandated ones) they're not even allowed to..

As far as anything stopping amazon paying to make the competitors go away.. only the risk of reputational damage when people inevitably found out, otherwise they'd have already done it.

**Comment by Roosterno1** [comment]
Seems nobody has been tasked to look at this... maybe someone needs to report it via Telegram  ??  (Which I dont use)
Shame, as its my fave app  :-(

**Comment by tidder99** [comment]
I also confirm that it looks like it's fixed

**Comment by tidder99** [comment]
Just installed a older version and it's still working so it might be Virgin media who has fixed something

**Comment by danpilch** [comment]
Can confirm with VM most videos buffer and freeze constantly

**Comment by p-money666** [comment]
I recently moved to VM and everything was fine until a few updates ago when I've had really bad buffering.

I thought it was an app issue but now I'm not so sure.

I recently started using a VPN on my TV may need to try it without it and see if that helps.

Would have thought a VPN would improve things not the opposite but I'll see.

**Comment by p-money666** [comment]
> @TonyHoyle I dug through the code whilst trying to get it to run and debug the networking (was unsuccessful) however from what I could see it simply connects to YouTube via a normal http get request using YouTube.com/TV spoofed as a referrer, and a Samsung TV as the device.
> 
> I'm not an expert on networking, however, I'd be amazed if Virgin Media has managed to implement something to block/throttle what is essentially a http request masquerading as the YouTube app on a Samsung TV.
> 
> Really interesting you have it working via ipv6. Wasn't aware VM were handing those out (at least on my residential line, anyway!) Would be interested in hearing more about how you've managed that if you have the time to share?
> 
> 
> Update: Given up on this and just resubscribed to Mullvad. Decent VPN provider with a good app that supports split tunnelling (so Netflix etc will still work).
> 
> Smart Tube working fantastic again. Would recommend setting the buffer in Smart Tube settings to Medium or High if you go this route.
> 
> Once my contract is up with VM I'll drop them in favour of BT, not long until FTTP supposedly coming online in my area in 2023 and that is also when my contract ends with VM. This is just one incident in a list of debacles with them and quite frankly I can't be bothered to fight with them anymore.
> 
> Whilst I'm here - huge thanks to the Dev for creating this app. It is a godsend. 

I'm on VM and it's been perfect apart from this. This is the last issue I expected to have with them! And I'm not even sure if it's a buffering bug on the app (like countless others have reported) or if it's a VM issue (why on earth would they target this of all things? Are they that pro-Google/YouTube). I highly doubt it.

I'm using a VPN on my TV but I think it's got worse since using the VPN! Had no issues before that. But I'll have a tinker especially with the buffering settings and see how it goes.

Thanks.
